packages/json-api/src/Registry.ts

```diff
@@ -96,12 +96,32 @@ export default class RegistryImpl implements Registry {
         }, {})
       : undefined;
 
-    if (Array.isArray(document.data)) {
-      return (document.data as JSONAPIResource[]).map(elem =>
-        parseResource(elem, includesMap)
+    let included: { [key: string]: Array<JSONValue> } = {};
+    if (includesMap) {
+      included = Object.keys(includesMap).reduce(
+        (acc: { [key: string]: Array<JSONValue> }, key) => {
+          const unifiedKey: string = key.split(":")[0];
+          if (acc[unifiedKey]) acc[unifiedKey].push(includesMap[key]);
```

Comment:
There's something wrong here, because you're trying to undo what has been done 20 lines before. We can do it in the same place in which we build `includedMap`.

Reply:
Yes, I'll change it